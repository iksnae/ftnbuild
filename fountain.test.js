const fs = require('fs')
const fountain = require('./fountain')

describe('fountain', () => {
  it('should run against fixture', () => {
    const sample = fs.readFileSync('./samples/brick&steel.fountain', {
      encoding: 'utf-8',
    })
    expect(fountain.parse(sample, true)).toMatchSnapshot()
  })
})
