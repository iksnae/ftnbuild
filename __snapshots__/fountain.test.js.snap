// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fountain should run against fixture 1`] = `
Object {
  "html": Object {
    "script": "<h3>EXT. BRICK'S PATIO - DAY</h3><p>A gorgeous day.  The sun is shining.  But BRICK BRADDOCK, retired police detective, is sitting quietly, contemplating -- something.</p><p>The SCREEN DOOR slides open and DICK STEEL, his former partner and fellow retiree, emerges with two cold beers.</p><div class=\\"dialogue\\"><h4>STEEL</h4><p>Beer's ready!</p></div> <div class=\\"dialogue\\"><h4>BRICK</h4><p>Are they cold?</p></div> <div class=\\"dialogue\\"><h4>STEEL</h4><p>Does a bear crap in the woods?</p></div> <p>Steel sits.  They laugh at the dumb joke.</p><div class=\\"dialogue\\"><h4>STEEL</h4><p class=\\"parenthetical\\">(beer raised)</p><p>To retirement.</p></div> <div class=\\"dialogue\\"><h4>BRICK</h4><p>To retirement.</p></div> <p>They drink long and well from the beers.</p><p>And then there's a long beat.<br />Longer than is funny.<br />Long enough to be depressing.</p><p>The men look at each other.</p><div class=\\"dual-dialogue\\"><div class=\\"dialogue left\\"><h4>STEEL</h4><p>Screw retirement.</p></div> <div class=\\"dialogue right\\"><h4>BRICK</h4><p>Screw retirement.</p></div> </div> <h2>SMASH CUT TO:</h2><h3>INT. TRAILER HOME - DAY</h3><p>This is the home of THE BOY BAND, AKA DAN and JACK.  They too are drinking beer, and counting the take from their last smash-and-grab.  Money, drugs, and ridiculous props are strewn about the table.</p><div class=\\"dialogue\\"><h4>JACK</h4><p class=\\"parenthetical\\">(in Vietnamese, subtitled)</p><p><span class=\\"italic\\">Did you know Brick and Steel are retired?</span></p></div> <div class=\\"dialogue\\"><h4>DAN</h4><p>Then let's retire them.<br /><span class=\\"underline\\">Permanently</span>.</p></div> <p>Jack begins to argue vociferously in Vietnamese (?), But mercifully we...</p><h2>CUT TO:</h2><h3>EXT. BRICK'S POOL - DAY</h3><p>Steel, in the middle of a heated phone call:</p><div class=\\"dialogue\\"><h4>STEEL</h4><p>They're coming out of the woodwork!<br /></p><p class=\\"parenthetical\\">(pause)</p><p>No, everybody we've put away!<br /></p><p class=\\"parenthetical\\">(pause)</p><p>Point Blank Sniper?</p></div> <h3>SNIPER SCOPE POV</h3><p>From what seems like only INCHES AWAY.  <span class=\\"underline\\">Steel's face FILLS the <span class=\\"italic\\">Leupold Mark 4</span> scope</span>.</p><div class=\\"dialogue\\"><h4>STEEL</h4><p>The man's a myth!</p></div> <p>Steel turns and looks straight into the cross-hairs.</p><div class=\\"dialogue\\"><h4>STEEL</h4><p class=\\"parenthetical\\">(oh crap)</p><p>Hello...</p></div> <h2>CUT TO:</h2><h3>OPENING TITLES</h3><p class=\\"centered\\">BRICK BRADDOCK <br /> & DICK STEEL IN</p><p class=\\"centered\\">BRICK & STEEL <br /> FULL RETIRED</p><h2>SMASH CUT TO:</h2><h3>EXT. WOODEN SHACK - DAY</h3><p>COGNITO, the criminal mastermind, is SLAMMED against the wall.</p><div class=\\"dialogue\\"><h4>COGNITO</h4><p>Woah woah woah, Brick and Steel!</p></div> <p>Sure enough, it's Brick and Steel, roughing up their favorite usual suspect.</p><div class=\\"dialogue\\"><h4>COGNITO</h4><p>What is it you want with me, DICK?</p></div> <p>Steel SMACKS him.</p><div class=\\"dialogue\\"><h4>STEEL</h4><p>Who's coming after us?</p></div> <div class=\\"dialogue\\"><h4>COGNITO</h4><p>Everyone's coming after you mate!  Scorpio, The Boy Band, Sparrow, Point Blank Sniper...</p></div> <p>As he rattles off the long list, Brick and Steel share a look.  This is going to be BAD.</p><h2>CUT TO:</h2><h3>INT. GARAGE - DAY</h3><p>BRICK and STEEL get into Mom's PORSCHE, Steel at the wheel.  They pause for a beat, the gravity of the situation catching up with them.</p><div class=\\"dialogue\\"><h4>BRICK</h4><p>This is everybody we've ever put away.</p></div> <div class=\\"dialogue\\"><h4>STEEL</h4><p class=\\"parenthetical\\">(starting the engine)</p><p>So much for retirement!</p></div> <p>They speed off.  To destiny!</p><h2>CUT TO:</h2><h3>EXT. PALATIAL MANSION - DAY</h3><p>An EXTREMELY HANDSOME MAN drinks a beer.  Shirtless, unfortunately.</p><p>His minion approaches offscreen:</p><div class=\\"dialogue\\"><h4>MINION</h4><p>We found Brick and Steel!</p></div> <div class=\\"dialogue\\"><h4>HANDSOME MAN</h4><p>I want them dead.  DEAD!</p></div> <p>Beer flies.</p><h2>BURN TO PINK.</h2><p class=\\"centered\\">THE END</p>",
    "title_page": "<h1><span class=\\"bold underline\\">BRICK & STEEL</span><br /><span class=\\"bold underline\\">FULL RETIRED</span></h1><p class=\\"credit\\">Written by</p><p class=\\"authors\\">Stu Maschwitz</p><p class=\\"source\\">Story by KTM</p><p class=\\"draft-date\\">1/27/2012</p><p class=\\"contact\\">Next Level Productions<br />1588 Mission Dr.<br />Solvang, CA 93463</p>",
  },
  "title": "BRICK & STEEL FULL RETIRED",
  "tokens": Array [
    Object {
      "text": "<span class=\\"bold underline\\">BRICK & STEEL</span><br /><span class=\\"bold underline\\">FULL RETIRED</span>",
      "type": "title",
    },
    Object {
      "text": "Written by",
      "type": "credit",
    },
    Object {
      "text": "Stu Maschwitz",
      "type": "author",
    },
    Object {
      "text": "Story by KTM",
      "type": "source",
    },
    Object {
      "text": "1/27/2012",
      "type": "draft_date",
    },
    Object {
      "text": "Next Level Productions<br />1588 Mission Dr.<br />Solvang, CA 93463",
      "type": "contact",
    },
    Object {
      "scene_number": undefined,
      "text": "EXT. BRICK'S PATIO - DAY",
      "type": "scene_heading",
    },
    Object {
      "text": "A gorgeous day.  The sun is shining.  But BRICK BRADDOCK, retired police detective, is sitting quietly, contemplating -- something.",
      "type": "action",
    },
    Object {
      "text": "The SCREEN DOOR slides open and DICK STEEL, his former partner and fellow retiree, emerges with two cold beers.",
      "type": "action",
    },
    Object {
      "dual": undefined,
      "text": undefined,
      "type": "dialogue_begin",
    },
    Object {
      "text": "STEEL",
      "type": "character",
    },
    Object {
      "text": "Beer's ready!",
      "type": "dialogue",
    },
    Object {
      "text": undefined,
      "type": "dialogue_end",
    },
    Object {
      "dual": undefined,
      "text": undefined,
      "type": "dialogue_begin",
    },
    Object {
      "text": "BRICK",
      "type": "character",
    },
    Object {
      "text": "Are they cold?",
      "type": "dialogue",
    },
    Object {
      "text": undefined,
      "type": "dialogue_end",
    },
    Object {
      "dual": undefined,
      "text": undefined,
      "type": "dialogue_begin",
    },
    Object {
      "text": "STEEL",
      "type": "character",
    },
    Object {
      "text": "Does a bear crap in the woods?",
      "type": "dialogue",
    },
    Object {
      "text": undefined,
      "type": "dialogue_end",
    },
    Object {
      "text": "Steel sits.  They laugh at the dumb joke.",
      "type": "action",
    },
    Object {
      "dual": undefined,
      "text": undefined,
      "type": "dialogue_begin",
    },
    Object {
      "text": "STEEL",
      "type": "character",
    },
    Object {
      "text": "(beer raised)",
      "type": "parenthetical",
    },
    Object {
      "text": "To retirement.",
      "type": "dialogue",
    },
    Object {
      "text": undefined,
      "type": "dialogue_end",
    },
    Object {
      "dual": undefined,
      "text": undefined,
      "type": "dialogue_begin",
    },
    Object {
      "text": "BRICK",
      "type": "character",
    },
    Object {
      "text": "To retirement.",
      "type": "dialogue",
    },
    Object {
      "text": undefined,
      "type": "dialogue_end",
    },
    Object {
      "text": "They drink long and well from the beers.",
      "type": "action",
    },
    Object {
      "text": "And then there's a long beat.<br />Longer than is funny.<br />Long enough to be depressing.",
      "type": "action",
    },
    Object {
      "text": "The men look at each other.",
      "type": "action",
    },
    Object {
      "text": undefined,
      "type": "dual_dialogue_begin",
    },
    Object {
      "dual": "left",
      "text": undefined,
      "type": "dialogue_begin",
    },
    Object {
      "text": "STEEL",
      "type": "character",
    },
    Object {
      "text": "Screw retirement.",
      "type": "dialogue",
    },
    Object {
      "text": undefined,
      "type": "dialogue_end",
    },
    Object {
      "dual": "right",
      "text": undefined,
      "type": "dialogue_begin",
    },
    Object {
      "text": "BRICK",
      "type": "character",
    },
    Object {
      "text": "Screw retirement.",
      "type": "dialogue",
    },
    Object {
      "text": undefined,
      "type": "dialogue_end",
    },
    Object {
      "text": undefined,
      "type": "dual_dialogue_end",
    },
    Object {
      "text": "SMASH CUT TO:",
      "type": "transition",
    },
    Object {
      "scene_number": undefined,
      "text": "INT. TRAILER HOME - DAY",
      "type": "scene_heading",
    },
    Object {
      "text": "This is the home of THE BOY BAND, AKA DAN and JACK.  They too are drinking beer, and counting the take from their last smash-and-grab.  Money, drugs, and ridiculous props are strewn about the table.",
      "type": "action",
    },
    Object {
      "dual": undefined,
      "text": undefined,
      "type": "dialogue_begin",
    },
    Object {
      "text": "JACK",
      "type": "character",
    },
    Object {
      "text": "(in Vietnamese, subtitled)",
      "type": "parenthetical",
    },
    Object {
      "text": "<span class=\\"italic\\">Did you know Brick and Steel are retired?</span>",
      "type": "dialogue",
    },
    Object {
      "text": undefined,
      "type": "dialogue_end",
    },
    Object {
      "dual": undefined,
      "text": undefined,
      "type": "dialogue_begin",
    },
    Object {
      "text": "DAN",
      "type": "character",
    },
    Object {
      "text": "Then let's retire them.<br /><span class=\\"underline\\">Permanently</span>.",
      "type": "dialogue",
    },
    Object {
      "text": undefined,
      "type": "dialogue_end",
    },
    Object {
      "text": "Jack begins to argue vociferously in Vietnamese (?), But mercifully we...",
      "type": "action",
    },
    Object {
      "text": "CUT TO:",
      "type": "transition",
    },
    Object {
      "scene_number": undefined,
      "text": "EXT. BRICK'S POOL - DAY",
      "type": "scene_heading",
    },
    Object {
      "text": "Steel, in the middle of a heated phone call:",
      "type": "action",
    },
    Object {
      "dual": undefined,
      "text": undefined,
      "type": "dialogue_begin",
    },
    Object {
      "text": "STEEL",
      "type": "character",
    },
    Object {
      "text": "They're coming out of the woodwork!<br />",
      "type": "dialogue",
    },
    Object {
      "text": "(pause)",
      "type": "parenthetical",
    },
    Object {
      "text": "No, everybody we've put away!<br />",
      "type": "dialogue",
    },
    Object {
      "text": "(pause)",
      "type": "parenthetical",
    },
    Object {
      "text": "Point Blank Sniper?",
      "type": "dialogue",
    },
    Object {
      "text": undefined,
      "type": "dialogue_end",
    },
    Object {
      "scene_number": undefined,
      "text": "SNIPER SCOPE POV",
      "type": "scene_heading",
    },
    Object {
      "text": "From what seems like only INCHES AWAY.  <span class=\\"underline\\">Steel's face FILLS the <span class=\\"italic\\">Leupold Mark 4</span> scope</span>.",
      "type": "action",
    },
    Object {
      "dual": undefined,
      "text": undefined,
      "type": "dialogue_begin",
    },
    Object {
      "text": "STEEL",
      "type": "character",
    },
    Object {
      "text": "The man's a myth!",
      "type": "dialogue",
    },
    Object {
      "text": undefined,
      "type": "dialogue_end",
    },
    Object {
      "text": "Steel turns and looks straight into the cross-hairs.",
      "type": "action",
    },
    Object {
      "dual": undefined,
      "text": undefined,
      "type": "dialogue_begin",
    },
    Object {
      "text": "STEEL",
      "type": "character",
    },
    Object {
      "text": "(oh crap)",
      "type": "parenthetical",
    },
    Object {
      "text": "Hello...",
      "type": "dialogue",
    },
    Object {
      "text": undefined,
      "type": "dialogue_end",
    },
    Object {
      "text": "CUT TO:",
      "type": "transition",
    },
    Object {
      "scene_number": undefined,
      "text": "OPENING TITLES",
      "type": "scene_heading",
    },
    Object {
      "text": "BRICK BRADDOCK <br /> & DICK STEEL IN",
      "type": "centered",
    },
    Object {
      "text": "BRICK & STEEL <br /> FULL RETIRED",
      "type": "centered",
    },
    Object {
      "text": "SMASH CUT TO:",
      "type": "transition",
    },
    Object {
      "scene_number": undefined,
      "text": "EXT. WOODEN SHACK - DAY",
      "type": "scene_heading",
    },
    Object {
      "text": "COGNITO, the criminal mastermind, is SLAMMED against the wall.",
      "type": "action",
    },
    Object {
      "dual": undefined,
      "text": undefined,
      "type": "dialogue_begin",
    },
    Object {
      "text": "COGNITO",
      "type": "character",
    },
    Object {
      "text": "Woah woah woah, Brick and Steel!",
      "type": "dialogue",
    },
    Object {
      "text": undefined,
      "type": "dialogue_end",
    },
    Object {
      "text": "Sure enough, it's Brick and Steel, roughing up their favorite usual suspect.",
      "type": "action",
    },
    Object {
      "dual": undefined,
      "text": undefined,
      "type": "dialogue_begin",
    },
    Object {
      "text": "COGNITO",
      "type": "character",
    },
    Object {
      "text": "What is it you want with me, DICK?",
      "type": "dialogue",
    },
    Object {
      "text": undefined,
      "type": "dialogue_end",
    },
    Object {
      "text": "Steel SMACKS him.",
      "type": "action",
    },
    Object {
      "dual": undefined,
      "text": undefined,
      "type": "dialogue_begin",
    },
    Object {
      "text": "STEEL",
      "type": "character",
    },
    Object {
      "text": "Who's coming after us?",
      "type": "dialogue",
    },
    Object {
      "text": undefined,
      "type": "dialogue_end",
    },
    Object {
      "dual": undefined,
      "text": undefined,
      "type": "dialogue_begin",
    },
    Object {
      "text": "COGNITO",
      "type": "character",
    },
    Object {
      "text": "Everyone's coming after you mate!  Scorpio, The Boy Band, Sparrow, Point Blank Sniper...",
      "type": "dialogue",
    },
    Object {
      "text": undefined,
      "type": "dialogue_end",
    },
    Object {
      "text": "As he rattles off the long list, Brick and Steel share a look.  This is going to be BAD.",
      "type": "action",
    },
    Object {
      "text": "CUT TO:",
      "type": "transition",
    },
    Object {
      "scene_number": undefined,
      "text": "INT. GARAGE - DAY",
      "type": "scene_heading",
    },
    Object {
      "text": "BRICK and STEEL get into Mom's PORSCHE, Steel at the wheel.  They pause for a beat, the gravity of the situation catching up with them.",
      "type": "action",
    },
    Object {
      "dual": undefined,
      "text": undefined,
      "type": "dialogue_begin",
    },
    Object {
      "text": "BRICK",
      "type": "character",
    },
    Object {
      "text": "This is everybody we've ever put away.",
      "type": "dialogue",
    },
    Object {
      "text": undefined,
      "type": "dialogue_end",
    },
    Object {
      "dual": undefined,
      "text": undefined,
      "type": "dialogue_begin",
    },
    Object {
      "text": "STEEL",
      "type": "character",
    },
    Object {
      "text": "(starting the engine)",
      "type": "parenthetical",
    },
    Object {
      "text": "So much for retirement!",
      "type": "dialogue",
    },
    Object {
      "text": undefined,
      "type": "dialogue_end",
    },
    Object {
      "text": "They speed off.  To destiny!",
      "type": "action",
    },
    Object {
      "text": "CUT TO:",
      "type": "transition",
    },
    Object {
      "scene_number": undefined,
      "text": "EXT. PALATIAL MANSION - DAY",
      "type": "scene_heading",
    },
    Object {
      "text": "An EXTREMELY HANDSOME MAN drinks a beer.  Shirtless, unfortunately.",
      "type": "action",
    },
    Object {
      "text": "His minion approaches offscreen:",
      "type": "action",
    },
    Object {
      "dual": undefined,
      "text": undefined,
      "type": "dialogue_begin",
    },
    Object {
      "text": "MINION",
      "type": "character",
    },
    Object {
      "text": "We found Brick and Steel!",
      "type": "dialogue",
    },
    Object {
      "text": undefined,
      "type": "dialogue_end",
    },
    Object {
      "dual": undefined,
      "text": undefined,
      "type": "dialogue_begin",
    },
    Object {
      "text": "HANDSOME MAN",
      "type": "character",
    },
    Object {
      "text": "I want them dead.  DEAD!",
      "type": "dialogue",
    },
    Object {
      "text": undefined,
      "type": "dialogue_end",
    },
    Object {
      "text": "Beer flies.",
      "type": "action",
    },
    Object {
      "text": "BURN TO PINK.",
      "type": "transition",
    },
    Object {
      "text": "THE END",
      "type": "centered",
    },
  ],
}
`;
